linker/shared/src/main/scala/org/scalajs/linker/checker/ClassDefChecker.scala

```diff
@@ -299,11 +299,17 @@ private final class ClassDefChecker(classDef: ClassDef,
     }
 
     // Body
-    val thisType = if (static) NoType else instanceThisType
-    val bodyEnv = Env.fromParams(params)
-      .withThisType(thisType)
-      .withInConstructor(isConstructor)
-    body.foreach(checkTree(_, bodyEnv))
+    for (body <- optBody) {
+      val thisType = if (static) NoType else instanceThisType
+      val bodyEnv = Env.fromParams(params)
+        .withThisType(thisType)
+        .withInConstructor(isConstructor)
```

Comment:
Does it make sense (maybe in a follow-up commit or even PR) to remove this env value and check `StoreModule` inside `checkConstructorBody`? (I realize this probably makes presence of the tree more restrictive).

Reply:
I think that would make sense, but I believe it belongs more to #5019 than this PR.